.github/workflows/ci.yml

```diff
@@ -80,37 +100,48 @@ jobs:
         uses: actions/setup-python@v2
         with:
           python-version: 3.8
-      - name: Install MSRV Rust
+      - name: Install Rust
         uses: actions-rs/toolchain@v1
         with:
-          profile: minimal
           toolchain: 1.48.0
+          profile: minimal
           default: true
-      - name: Install maturin, poetry, and toml
-        run: pip install maturin poetry toml
-      - name: Create an isolated example directory
-        run: cp -r examples/simple-extension/ ../simple-extension-msrv
-      - name: Edit Cargo.toml and change the path of rust-numpy
+      - uses: Swatinem/rust-cache@v1
+        with:
+          working-directory: examples/simple-extension
+        continue-on-error: true
+      - name: Install toml
+        run: pip install toml
+      - name: Edit Cargo.toml and detach from workspace
         run: |
           import toml
           cargo_toml = toml.load("Cargo.toml")
-          cargo_toml["dependencies"]["numpy"]["path"] = "../rust-numpy"
           cargo_toml["dependencies"]["ndarray"] = "0.13.1"
           cargo_toml["dependencies"]["num-complex"] = "0.2.4"
+          cargo_toml["workspace"] = {}
```

Comment:
I like this 👍🏼

Reply:
Me too, but it was almost a necessity due to caching: The caching action would fail if its working directory was not present when run, hence building within an existing directory significantly simplified things compared to re-ordering steps so that the copy was created first, and then the caching action triggered.